### PR TITLE
MVP liquidity pool

### PR DESCRIPTION
**Pull request by asardon**
* added simple LP for the MVP with basic fundraising functionalities

### CODE REVIEW

File: src/test/LiquidityPool.spec.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  });
+
+  it("can let users redeem LP shares", async () => {
+    var funder_balance = new BN(await USDC_contract.methods.balanceOf(address_with_USDC).call());
+
+    var LP_balance = new BN(await USDC_contract.methods.balanceOf(liquidityPool.address).call());
+    console.log("LP_balance :" + LP_balance);
+    expect(LP_balance.gt(new BN(0))).to.be.true;
+
+    await liquidityPool.redeem({from: address_with_USDC});
```
Comment (xh3b4sd) [inline src/test/LiquidityPool.spec.ts:125]: I don't understand how the address with USDC can redeem anything when in this test case it never funded the LP. I am kind of missing something here.

File: src/contracts/LiquidityPool.sol [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    address public constant USDC_addr = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    uint256 public fund_raising_start;
+    uint256 public provisioning_start;
+    uint256 public constant fund_raising_duration = 2419200; // 28 days
+    uint256 public constant provisioning_period = 7776000; // 90 days
+    uint256 public constant minimum_funding_amount = 10000000 * 10 ** 6; // 10 million in USDC, with 6 decimals
+    bool public fund_raising_finalized;
+    bool public fund_raising_success;
+    bool public funds_deployed;
+    bool public is_redeemable;
```
[the remark this answers is not in the document]
Comment (asardon) [reply src/contracts/LiquidityPool.sol:22]: Thx! Good point, noted.

File: src/contracts/LiquidityPool.sol [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        IERC20(USDC_addr).approve(AMM_addr, usdc_balance);
+        AutomatedMarketMaker(AMM_addr).stake(usdc_balance);
+        provisioning_start = block.timestamp;
+        funds_deployed = true;
+    }
+
+    function withdraw_funds() public {
+        require(funds_deployed, "funds haven´t been deployed");
+        require(provisioning_start + provisioning_period < block.timestamp, "LP is still in provisioning phase");
+        AutomatedMarketMaker(AMM_addr).unstake_all();
```
[the remark this answers is not in the document]
Comment (asardon) [reply src/contracts/LiquidityPool.sol:57]: Thx for your comment! After the LP has withdrawn all funds, users can start redeeming their share on a pro rata basis.

File: src/test/LiquidityPool.spec.ts (hunk 1 above)
Comment (asardon) [reply src/test/LiquidityPool.spec.ts:125]: Thx for your comment, the `it("...")` tests should run consecutively. And previously the `address_with_USDC` funded the LP.

Comment (xh3b4sd) [reply src/test/LiquidityPool.spec.ts:125]: That helps my understanding of these JS tests quite a bit. I was under the impression they run in isolation.

**Pull request merged**